Transpostion Cipher - Encryption

rajeshkumargp opened the pull request:
encrypt function in transpos module is created.

rajeshkumargp committed:
Added encrypt() function in transpos module

The transposition cipher encryption/decryption is implemented in this
module.

rajeshkumargp committed:
Revert "Fix list formatting."

This reverts commit d0a5f4dfa45b8352f7e4fff7d46f28b1f0b851ea.

rajeshkumargp committed:
Revert "Revert "Fix list formatting.""

This reverts commit 05405f2f404408ceb1c7a85984e8bd49d16a29e9.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,35 @@
+"""
+
+
+def encrypt(keysize,message):
```
Please run `pep8` tool on your code.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,35 @@
+def encrypt(keysize,message):
+    '''
+    Args :
+        Transposition Cipher Key Size,Message
```
Please read the Google docstring convention, mentioned in the contributing guidelines https://github.com/chennaipy/libcipher#contributing-guidelines

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,35 @@
+        encrypted_message = ''.join(encrypted_message)+'|'
+        return encrypted_message
+    else:
+        #print('Key Size is Invalid')       
```
Avoid code in comments.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,35 @@
+
+Provides functions to encrypt / decrypt a message by transpose ciphering it.
+
+Encryption needs a key(a number).
```
See below for the docstring format and layout.

```
"""Single line description.

Followed by a long description.
""""
```

rajeshkumargp committed:
Encryption Function is updated

Encryption function is coded to handle empty strings,and exception
thrown for InvalidKeySize.

rajeshkumargp committed:
Added Code after PEP8 verification

transpos.py  and transpos_test.py is updated with PEP8 standards.

rajeshkumargp commented:
Hi bravegnu i have updated code with pep8 standards. 
I need help on following
1. I have run pep8 for transpos module.I have fixed all but one regarding hanging indent.
2. By Googling,the code to test for Exception raising is written. But it is failing while execution. 
If possible provide any examples,I will go through and work on that.

whiletruelearn commented:
@rajeshkumargp I am testing whether exceptions are raised in #10 . Maybe you could use that as an example.

rajeshkumargp committed:
Tranpos Cipher-Encryption

NoKeyGiven Exception is handled in transpos-encrypt function. Pep8
validation success for transpos.py and transpos_test.py.

rajeshkumargp commented:
@kskrishnasangeeth  Thanks for your support.The issue is fixed with help of example provided.

shrayasr commented:
Your tests are failing on Python 2.6 and 2.7. Please have a look

rajeshkumargp commented:
I have done compiling with Python 3. Do i need to do code the same for Python 2 also ? 
Can you please say What i need to do to fix that ?

shrayasr commented:
Yes you need to do it for python 2 as well. You can go over to the [details](https://travis-ci.org/Chennaipy/libcipher/builds/59206027) of this PR's tests on the Travis website.

rajeshkumargp committed:
Updated With Python2 and Python3 compatibility

Code is fixed for Python2 and Python3 compatibility,Empty Input message
is also handled by exception

rajeshkumargp commented:
With guidance of team, We have done with Python2 and Python3 compatible Code,Pep8 validated code for Transposition Cipher Encryption.

rajeshkumargp commented:
What to do next ? Is coding and pep8validation is sufficient?

shrayasr commented:
> What to do next ? Is coding and pep8validation is sufficient?

Lets wait for @bravegnu to review the code part. Since he started with it, its best if he finishes it

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,92 @@
+        message encrypted using transposition cipher.'|' is added at to
+        indicate end of message.For invalid key size,InvalidKeySizeException
+        is thrown.For empty input message,'|' is returned.
+    '''
```
The docstring needs to be fixed. Please see http://sphinxcontrib-napoleon.readthedocs.org/en/latest/example_google.html

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,92 @@
+        is thrown.For empty input message,'|' is returned.
+    '''
+    if message is None:
+        raise NoMessageGivenException
```
This check is not required.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,92 @@
+    # the transposition cipher key is limited to half the length of the
+    # message it is used to encrypt
+    if keysize is None:
+        raise NoKeyGivenException
```
Ditto.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,92 @@
+    if keysize is None:
+        raise NoKeyGivenException
+    else:
+        if keysize <= message_size//2:
```
Did you run your code through `pep8` tool?

rajeshkumargp replied:
The code is checked with PEP8 tool.
I will check the link provided and update docstring.
I want to get clarified on What would happen if no message is given as input ie. NoneType
Similarly the scenario for None KeySize.
Still If it is not required means,Why ?

bravegnu replied:
You might want to try reading http://stackoverflow.com/questions/19684434/best-way-to-check-function-arguments-in-python

rajeshkumargp committed:
Update Docstrings

Docstrings have been  updated.

rajeshkumargp commented:
The docstrings are updated with guidance of links provided.Can you please review it ?

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+    pass
+
+
+class InvalidKeySizeException(Error):
```
We could just use the `ValueError` exception instead of defining our own.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+    Returns:
+        String : For valid message and keysize,encrypetd message with '|'
+          at end is returned.
+          When message is '','|' is alone returned as encrypted message.
```
The `|` is not part of the encryption algorithm. It is added at the end of the string while printing so that the spaces at the end of the message is visible.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+
+
+def encrypt(keysize, message):
+    '''Transpose Cipher Encrypt function.
```
How about "Returns encrypted message."?

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+    message_size = len(message)
+    # the transposition cipher key is limited to half the length of the
+    # message it is used to encrypt
+    if keysize <= message_size//2:
```
It would better to check the key, and raise the exception. And de-dent the remaining code.

```
if key < message // 2:
    raise ValueError("Key size less than half of message size")
```

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+        while letter_index < message_size:
+            column_index = 0
+            while column_index < keysize and letter_index < message_size:
+                encrypted_message[column_index] += message[letter_index]
```
The original algorithm in the book looks simpler. Why don't we just use it. Nothing wrong in using the code.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+    # the transposition cipher key is limited to half the length of the
+    # message it is used to encrypt
+    if keysize <= message_size//2:
+        encrypted_message = keysize * ['']
```
`[''] * key` looks much better.

bravegnu commented on libcipher/transpos.py:
```diff
@@ -0,0 +1,77 @@
+            " and Message Size is " + repr(self.messagesize))
+
+
+def encrypt(keysize, message):
```
How about naming the argument `key` instead of `keysize`.

rajeshkumargp committed:
Variable renamed

Updated with '|' removed and variable renameing

rajeshkumargp committed:
Code is updated with variable renamed,comments updated

Updated Code as in Book

rajeshkumargp commented:
Hi @bravegnu The code is updated with variables renamed,'|' appending is removed and few changes. Can you please review it ?

rajeshkumargp commented:
Hi @bravegnu What need to improve ? Can you please review?
